go/saltpack/specs/saltpack_armor.md

```diff
@@ -0,0 +1,193 @@
+END ARMOR.
+
+## Spacing and Delimiting
+
+The decoder has two pre-processing steps:
+
+1. Drop input up to the first period and after the second period. (Note that
+   the second period in the example above comes right *before* "END ARMOR". The
+   final period is ignored.)
+2. Strip all whitespace.
+
+That gives the encoder some flexibility in formatting. The header and footer
+can be anything, as long as the header ends with a period and there's a period
+before the footer. Also the word and line spacing can be anything.
+
+Our encoder emits a space every 15 characters and a newline every 200 words.
+That's what felt nicest when we tried it with different terminals and messaging
+apps. But since our decoder doesn't rely on this spacing, we can change it in
+the future.
+
+Note that with the 62-character alphabet we've chosen, the underlying Just
+Numbers encoding will turn 32-byte input blocks into 43-character output
+blocks. This is unrelated to the 15-character visible word length in our
+output; those visible words boundaries go away when we strip whitespace.
+
+## The Just Numbers Encoding
+
+The Just Numbers encoding scheme converts a block of bytes into a block of
+characters in some alphabet. It's defined for any alphabet and block size. The
+goal of the scheme is to work exactly like base conversion in normal
+arithmetic. To encode a block of B bytes with an alphabet of length A:
+
+1. Compute C, the length of the output character block. C is the smallest
+   integer such that `256^B <= A^C`.
+2. Interpret the byte block as a big-endian unsigned integer. For blocks longer
+   than 8 bytes, this usually requires a bignum library.
+3. Convert that number into a list of C digits in base A, by repeatedly taking
+   the remainder mod A and then dividing by A. Order the digits from most to
+   least significant. There may be leading zeroes.
+4. Map the digits into the letters in the target alphabet.
+
+To decode a block of C characters with an alphabet of length A:
+
+1. Compute B, the length of the output byte block. B is the largest integer
+   such that `256^B <= A^C`.
+2. Using the alphabet, map the characters in the block into a list of base A
+   digits.
+3. Multiply those digits by successive powers of A, and sum them together into
+   a number, with the leftmost digit being the most significant.
+4. Serialize that number as a B-byte big-endian unsigned integer.
+
+Note that unlike Base64 and Base85, there is no special treatment of "short
+blocks". Any block of any length can be encoded. It's the responsibility of the
+caller to chunk input into fixed-length blocks, and to choose an alphabet and
+block size that make the packing efficient. (The efficiency of a given block
+size is `B/C`.)
+
```

Comment:
Say that some size blocks are illegal.  At the end of a document, you must always chose the minimally-sized output block to encode into it.